datafusion/physical-plan/src/joins/utils.rs

```
@@ -634,26 +635,25 @@ fn max_distinct_count(
             Some(stats.distinct_count.clone())
         }
         (_, Some(max), Some(min)) => {
-            let numeric_range = Interval::new(
-                IntervalBound::new(min.clone(), false),
-                IntervalBound::new(max.clone(), false),
-            )
-            .cardinality()
-            .ok()
-            .flatten()? as usize;
+            let Ok(numeric_range) = Interval::try_new(min.clone(), max.clone()) else {
```

You are ignoring error here. Instead you may return Result from this function